drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/wizard/column/plugins/BRLConditionColumnPlugin.java

```diff
@@ -0,0 +1,303 @@
+import org.drools.workbench.screens.guided.dtable.client.wizard.column.plugins.commons.AdditionalInfoPageInitializer;
+import org.drools.workbench.screens.guided.dtable.client.wizard.column.plugins.commons.BaseDecisionTableColumnPlugin;
+import org.drools.workbench.screens.guided.rule.client.editor.RuleModellerConfiguration;
+import org.drools.workbench.screens.guided.rule.client.editor.events.TemplateVariablesChangedEvent;
+import org.uberfire.ext.widgets.core.client.wizards.WizardPage;
+
+import static org.drools.workbench.screens.guided.dtable.client.wizard.column.pages.common.DecisionTableColumnViewUtils.nil;
+
+@Dependent
+public class BRLConditionColumnPlugin extends BaseDecisionTableColumnPlugin implements HasRuleModellerPage,
+                                                                                       HasAdditionalInfoPage,
+                                                                                       TemplateVariablesChangedEvent.Handler {
+
+    @Inject
+    private RuleModellerPage ruleModellerPage;
+
+    @Inject
+    private AdditionalInfoPage additionalInfoPage;
+
+    private BRLConditionColumn editingCol;
+
+    private Boolean ruleModellerPageCompleted = Boolean.FALSE;
+
+    private HandlerRegistration registration;
+
+    private RuleModel ruleModel = null;
+
+    @Override
+    public void init(final NewGuidedDecisionTableColumnWizard wizard) {
+        super.init(wizard);
+
+        setupEditingCol();
+        setupRuleModellerEvents();
+    }
+
+    @Override
+    public void onClose() {
+        super.onClose();
+
+        teardownRuleModellerEvents();
+    }
+
+    @Override
+    public String getTitle() {
+        return translate(GuidedDecisionTableErraiConstants.BRLConditionColumnPlugin_AddConditionBRL);
+    }
+
+    @Override
+    public List<WizardPage> getPages() {
+        return new ArrayList<WizardPage>() {{
+            add(ruleModellerPage);
+            add(additionalInfoPage());
+        }};
+    }
+
+    @Override
+    public Boolean generateColumn() {
+        if (nil(editingCol.getHeader())) {
+            Window.alert(translate(GuidedDecisionTableErraiConstants.BRLConditionColumnPlugin_YouMustEnterAColumnHeaderValueDescription));
+            return false;
+        }
+
+        if (!isHeaderUnique(editingCol.getHeader())) {
+            Window.alert(translate(GuidedDecisionTableErraiConstants.BRLConditionColumnPlugin_ThatColumnNameIsAlreadyInUsePleasePickAnother));
+            return false;
+        }
+
+        getDefinedVariables(getRuleModel());
+
+        editingCol.setDefinition(Arrays.asList(getRuleModel().lhs));
+        presenter.appendColumn(editingCol);
+
+        return true;
+    }
+
+    @Override
+    public Type getType() {
+        return Type.ADVANCED;
+    }
+
+    boolean isHeaderUnique(String header) {
+        for (CompositeColumn<?> cc : presenter.getModel().getConditions()) {
```

Comment:
I guess the old code did only check uniqueness for headers in their "group"... weird.. IDK why!! ;-)